share/lib/python/neuron/rxd/rxd.py

```diff
@@ -1019,20 +1038,13 @@ def _get_node_indices(species, region, sec3d, x3d, sec1d, x1d):
     elif x1d == 1 - sec1d.orientation():
         # the ending zero-volume node is the one after the last node
         # TODO: make this more efficient
-        index_1d = max([node._index for node in species.nodes(region)(sec1d)]) + 1
+        index_1d, vol1d = max([(node._index, node.volume) for node in 
+                                species.nodes(region)(sec1d)],
+                               key=lambda x: x[0])
+        index_1d + 1
```

Comment:
Yes, index_1d is one after the last node which has zero volume, so the volume is taken for the last node. If the orientation were reversed, I think it would fall under the first condition, so index_1d would be the first node with the corresponding volume. This changes doesn't alter the indices, just the volumes associated with them.